kernel/hardware/miniuart.cpp

```diff
@@ -0,0 +1,110 @@
+#include <cstddef>
+#include <cstdint>
+
+#include "gpio.hpp"
+#include "miniuart.hpp"
+#include "mmio.hpp"
+
+namespace MINI_UART {
```

Comment:
What do you think about using the name MiniUART?

I find it prettier and more consistent with the rest.